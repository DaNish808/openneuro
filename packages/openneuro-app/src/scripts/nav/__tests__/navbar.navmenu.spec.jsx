import React from 'react'
import { shallow, mount } from 'enzyme'
import NavMenu from '../navbar.navmenu.jsx'

jest.mock('../../user/user.store.js')
jest.mock('../../upload/upload.store.js')
// Need to mock the router because of withRouter in these components
jest.mock('react-router-dom', () => {
  return {
    NavLink: () => '',
    withRouter: component => component,
  }
})

describe('NavMenu', () => {
  it('renders successfully', () => {
    const wrapper = shallow(<NavMenu isLoggedIn={null} loading={false} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders without "eventKey" warnings', () => {
    const component = mount(<NavMenu isLoggedIn={null} loading={false} />)
    expect(component).toMatchSnapshot()
  })
})
