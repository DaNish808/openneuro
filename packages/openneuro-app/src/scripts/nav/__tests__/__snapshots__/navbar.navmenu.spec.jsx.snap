// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu renders successfully 1`] = `
<ul
  className="nav navbar-nav main-nav"
>
  <li
    className="link-dashboard"
  />
  <li
    className="link-public"
  >
    <NavLink
      className="nav-link"
      to="/public/datasets"
    >
      <span
        className="link-name"
      >
        Public Dashboard
      </span>
    </NavLink>
  </li>
  <li
    className="link-support"
  >
    <a
      className="nav-link"
      onClick={[Function]}
    >
      <span
        className="link-name"
      >
        Support
      </span>
    </a>
  </li>
  <li
    className="link-faq"
  >
    <NavLink
      className="nav-link"
      to="/faq"
    >
      <span
        className="link-name"
      >
        faq
      </span>
    </NavLink>
  </li>
  <li
    className="link-admin"
  />
  <li
    className="link-dashboard"
  />
</ul>
`;

exports[`NavMenu renders without "eventKey" warnings 1`] = `
<NavMenu
  isLoggedIn={null}
  loading={false}
>
  <ul
    className="nav navbar-nav main-nav"
  >
    <li
      className="link-dashboard"
    />
    <li
      className="link-public"
    >
      <NavLink
        className="nav-link"
        to="/public/datasets"
      />
    </li>
    <li
      className="link-support"
    >
      <a
        className="nav-link"
        onClick={[Function]}
      >
        <span
          className="link-name"
        >
          Support
        </span>
      </a>
    </li>
    <li
      className="link-faq"
    >
      <NavLink
        className="nav-link"
        to="/faq"
      />
    </li>
    <li
      className="link-admin"
    />
    <li
      className="link-dashboard"
    />
  </ul>
</NavMenu>
`;
